File: Backend/src/Modules/ReservationModule/Shared/Dtos/OpenRulesDto.cs

using src.Modules.ReservationModule.Domain.Entities;

namespace src.Modules.ReservationModule.Shared.Dtos;

public class OpenRulesDto
{
    public required DateTime DefaultOpenDate { get; set; }
    public required DateTime DefaultCloseDate { get; set; }
    public required List<OpenTimeForDayDto> OpenTimesSingleDays { get; set; }
    public required List<WeekDayTimeSlotDto>DefaultOpenTimesForWeek { get; set; } 
}          

File: Frontend/web/src/models/openRulesDto.ts
import {WeeklyScheduleDto} from "./WeeklyScheduleDto.ts";
import {OpenTimeForDayDto} from "./OpenTimeForDayDto.ts";

export interface OpenRulesDto {
    defaultOpenDate: Date;
    defaultCloseDate: Date;
    openTimesSingleDays: OpenTimeForDayDto[];
    defaultOpenTimesForWeek: WeeklyScheduleDto;
}